plugins/autoturn.koplugin/main.lua

```diff
@@ -139,20 +141,29 @@ end
 function AutoTurn:addToMainMenu(menu_items)
     menu_items.autoturn = {
         sorting_hint = "navi",
-        text_func = function() return self:_enabled() and T(_("Autoturn: %1 s"), self.autoturn_sec)
-            or _("Autoturn") end,
+        text_func = function()
+            local time_string = util.secondsToClockDuration("modern", self.autoturn_sec, false, true, true, true)
+            return self:_enabled() and T(_("Autoturn: %1"), time_string) or _("Autoturn")
+        end,
         checked_func = function() return self:_enabled() end,
         callback = function(menu)
-            local SpinWidget = require("ui/widget/spinwidget")
-            local curr_items = G_reader_settings:readSetting("autoturn_timeout_seconds") or 30
-            local autoturn_spin = SpinWidget:new {
-                value = curr_items,
-                value_min = 0,
-                value_max = 240,
-                value_hold_step = 5,
+            local DateTimeWidget = require("ui/widget/datetimewidget")
+            local autoturn_seconds = G_reader_settings:readSetting("autoturn_timeout_seconds", 30)
+            local autoturn_minutes = math.floor(autoturn_seconds / 60)
+            autoturn_seconds = autoturn_seconds % 60
+            local autoturn_spin = DateTimeWidget:new {
+                is_date = false,
+                title_text = _("Autoturn time"),
+                info_text = _("Enter time in minutes and seconds."),
+                -- We use hour for minutes and minutes for seconds
```

Comment:
This is admittedly confusing, but I'm not really familiar with DTW, so I can't really say if there'd be a better approach...

Reply:
It's not more confusing than the `pickTimeoutValue` in autosuspend. When I combined datewidget and timewidget to datetimewidgetI never thought we would need seconds in there (as at this time it was not necessary). 
Now we need it in autosuspend and here (autoturn). 

Using not hh:mm but mm:ss is almost the same (difference the left value does not wrap at 24) -> So I am raping the DTW by renaming mm->ss and hh ->mm.

This PR is working fine, but as we are doing the trick hh:mm -> mm:ss now twice, it might be a good idea to drill up DTW to be able to handle hh:mm:ss (and optionally disable the hh or ss, or even disable the hh:mm or mm:ss). 
But in a follow up PR as then autoturn and autosuspend will be affected.